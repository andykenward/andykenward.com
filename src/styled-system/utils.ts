import shouldForwardProp from "@styled-system/should-forward-prop"
import { equals, indexOf } from "ramda"

import { colorPropsKeys } from "./colors"
import { fontSizePropsKeys } from "./fonts"

const propKeys = [...colorPropsKeys, ...fontSizePropsKeys]

export const filterProps: (prop: string) => boolean = (prop: string): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propKeys), -1)
