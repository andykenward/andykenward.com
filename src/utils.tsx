import shouldForwardProp from "@styled-system/should-forward-prop"
import { ThemeProvider } from "emotion-theming"
import { equals, indexOf } from "ramda"
import React from "react"
import { render } from "@testing-library/react"

import { theme } from "./theme"

export const filterProps: (prop: string, propsToFilter: string[]) => boolean = (
  prop: string,
  propsToFilter: string[]
): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propsToFilter), -1)

export const childrenOnly: (prop: string) => boolean = (
  prop: string
): boolean => prop === "children" || prop === "data-test-id"

export const renderWithTheme = (element: React.ReactElement<any>) =>
  render(<ThemeProvider theme={theme}>{element}</ThemeProvider>)
