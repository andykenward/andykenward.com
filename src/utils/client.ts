import { GraphQLClient } from "graphql-request"

import { getSdk } from "../../generated/graphql"

const graphqlClient = new GraphQLClient(process.env.API_URL, {
  headers: {
    authorization: `Bearer ${process.env.API_TOKEN}`,
  },
})

export const appSdk = (() => getSdk(graphqlClient))()
