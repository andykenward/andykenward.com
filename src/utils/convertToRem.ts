import { is } from "ramda"
/**
 *
 * UTILS
 */

export const convertToRem = (n: number | string): string | number =>
  is(Number, n) && n != 0 ? `${(n as number) / 16}rem` : n
