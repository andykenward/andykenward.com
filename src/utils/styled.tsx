import { SerializedStyles } from "@emotion/css"
import styled, { CreateStyled, CSSObject } from "@emotion/styled"
import React from "react"

import { Theme } from "../theme"

/**
 * TODO: pass `as` typings down if component
 */
export interface StyledProps {
  as?: keyof JSX.IntrinsicElements | React.ComponentType<any>
  css?: CSSObject | SerializedStyles
}

export default styled as CreateStyled<Theme>
