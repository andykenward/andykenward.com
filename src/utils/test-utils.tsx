import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { ThemeProvider } from "theme-ui"

import { theme } from "../theme"

const AllTheProviders: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllTheProviders, ...options })

export * from "@testing-library/react"

export { customRender as render }
