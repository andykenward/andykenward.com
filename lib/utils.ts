import shouldForwardProp from "@styled-system/should-forward-prop"
import { equals, indexOf } from "ramda"

import { propKeys } from "./styled-system"

export const filterProps: (prop: string) => boolean = (prop: string): boolean =>
  shouldForwardProp(prop) && equals(indexOf(prop, propKeys), -1)

export const childrenOnly: (prop: string) => boolean = (
  prop: string
): boolean => prop === "children" ?? prop === "data-test-id"
