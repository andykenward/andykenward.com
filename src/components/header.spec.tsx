import React from "react"

import { renderWithTheme } from "../utils"

import Header from "./header"

describe("Header", () => {
  it("renders correctly", () => {
    const { asFragment } = renderWithTheme(
      <Header siteTitle="Default Starter" />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
