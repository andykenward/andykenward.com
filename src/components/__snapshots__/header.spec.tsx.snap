// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<DocumentFragment>
  .emotion-3 {
  background-color: rebeccapurple;
}

.emotion-2 {
  font-size: 14px;
  margin-top: 0;
  margin-bottom: 0;
  margin-left: auto;
  margin-right: auto;
}

.emotion-0 {
  color: #fff;
}

<header
    class="emotion-3"
  >
    <div
      class="emotion-2"
    >
      <h1
        class="emotion-1"
      >
        <a
          class="emotion-0"
          href="/"
        >
          Default Starter
        </a>
      </h1>
    </div>
  </header>
</DocumentFragment>
`;
